storage-operator/pkg/controller/volume/volume_controller.go

```diff
@@ -88,85 +498,217 @@ type ReconcileVolume struct {
 // Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
 func (r *ReconcileVolume) Reconcile(request reconcile.Request) (reconcile.Result, error) {
 	reqLogger := log.WithValues("Request.Name", request.Name)
-	reqLogger.Info("Reconciling Volume")
+	reqLogger.Info("reconciling volume: START")
+	defer reqLogger.Info("reconciling volume: STOP")
 
-	// Fetch the Volume instance
-	instance := &storagev1alpha1.Volume{}
-	err := r.client.Get(context.TODO(), request.NamespacedName, instance)
+	ctx, cancel := context.WithCancel(context.TODO())
+	defer cancel()
+
+	// Fetch the requested Volume object.
+	volume := &storagev1alpha1.Volume{}
+	err := r.client.Get(ctx, request.NamespacedName, volume)
```

Comment:
I believe we also register watching `PersistentVolume`s owned by a `Volume`, so in this loop the `request` could either be about a `Volume` or a `PersistentVolume`.

I don't think this is an issue, because the lifetime of a `Volume`-owned `PersistentVolume` should always span the lifetime of the `Volume`, but it may be worth mentioning this in a code comment.

Reply:
Indeed, it's worth a comment.

Done.